locals.tf

```diff
@@ -1,6 +1,14 @@
 locals {
-  resource_name_pattern       = "spectral-${var.integration_type}-integration-${var.environment}"
+  resource_name_pattern       = coalesce(var.resource_name_common_part, "spectral-${var.integration_type}-integration-${var.environment}-${random_string.random_resource_name_suffix.id}")
   single_lambda_integration   = contains(["jira", "terraform"], var.integration_type) ? true : false
-  multiple_lambda_integration = contains(["gitlab"], var.integration_type) ? true : false
+  multiple_lambda_integration = contains(["gitlab", "github"], var.integration_type) ? true : false
   api_triggered_function_arn  = local.single_lambda_integration ? module.lambda_function[0].lambda_function_arn : module.frontend_lambda_function[0].lambda_function_arn
+  frontend_lambda_handler     = contains(["github"], var.integration_type) ? "index.handler" : "frontend.app"
+  backend_lambda_handler      = contains(["github"], var.integration_type) ? "index.handler" : "backend.app"
+  default_secrets_names = {
+    "github" = coalesce(var.secrets_names, ["Spectral_GithubBot_PrivateKey", "Spectral_GithubBot_WebhookSecret"]),
+    "gitlab" = coalesce(var.secrets_names, ["Spectral_GitlabBot_GitlabToken", "Spectral_GitlabBot_WebhookSecret"])
+  }
+  # Please do not change or replace the 'frontend' suffix since there a logic in the bot based in it
+  function_name = local.single_lambda_integration ? local.resource_name_pattern : "${local.resource_name_pattern}-frontend"
```

Comment:
`function_name` can be confusing since it implies working with both the backend and the front end. I recommend the following changes:

Rename options:
- receive_hook_lambda_name
- provider_request_treigger_lambda_name

The idea is to understand when you read this variable what he does.

Reply:
Changed to `api_integrated_function_name`